aws_jobs/cyhy-data-extract.py

```diff
@@ -133,39 +134,62 @@ def query_data(collection, query, tbz_file, tbz_filename, end_of_data_collection
     )
     collection_file = open(json_filename, "w")
     # How many documents into a query that will be skipped
-    skips = 0
-    # Number of documents in a query
-    count = collection.find(query, {"key": False}).count()
-    while skips < count:
-        # Pull documents between n and n + 100000
-        collection_file.write(
-            to_json(
-                list(
-                    collection.find(query, {"key": False}).skip(skips).limit(PAGE_SIZE)
-                )
-            )
-        )
-        skips += PAGE_SIZE
+    # skips = 0
+    # # Number of documents in a query
+    # count = collection.find(query, {"key": False}).count()
+    # while skips < count:
+    #     # Pull documents between n and n + 100000
+    #     collection_file.write(
+    #         to_json(
+    #             list(
+    #                 collection.find(query, {"key": False}).skip(skips).limit(PAGE_SIZE)
+    #             )
+    #         )
+    #     )
+    #     skips += PAGE_SIZE
+    # collection_file.close()
+    # if count > PAGE_SIZE:
+    #     # The first sed removes the ][ created by chunking the queries
+    #     # then the 2nd sed adds , to the document at the end of a
+    #     # chunked list
+    #     # If on MAC you will need gsed and gawk.
+    #     os.system(
+    #         r'sed -i "/\]\[/d" {} ; sed -i "s/\}}$/\}}\,/g" {} ; '.format(
+    #             json_filename, json_filename
+    #         )
+    #     )
+    #     # The previous sed will leave a }, at the last document in the
+    #     # list which is removed with this aws_access_key_id
+    #     os.system(
+    #         r"""awk 'NR==FNR{{tgt=NR-1;next}} (FNR==tgt) && /\}},/ {{ $1="    }}" }} 1' {} {} > {}.bak""".format(
+    #             json_filename, json_filename, json_filename
+    #         )
+    #     )
+    #     # This is a workaround for inplace
+    #     os.system("mv {}.bak {}".format(json_filename, json_filename))
+
+    collection_file.write("[")
+    last_id = None
+    while True:
+        if last_id == None:
+            result = collection.find(query, {'key': False}, sort=[('_id', pymongo.ASCENDING)]).limit(PAGE_SIZE)
+        else:
+            query['_id'] = {'$gt': last_id}
+            result = collection.find(query, {'key': False}, sort=[('_id', pymongo.ASCENDING)]).limit(PAGE_SIZE)
+
+        for doc in result:
+            last_id = doc['_id']
```

Comment:
`last_id = doc['_id']` can be moved to just after the end of the `for` loop so it is only done once at the end, rather than every time.